File: CODEVITA  TCS/Goodstring.java
import java.util.Scanner;

public class Goodstring {
    public static int totalDist(String name, String GoodStr) {
        int Distance = 0;
        char PrevGood = GoodStr.charAt(0);

        for (char charName : name.toCharArray()) {
            if (GoodStr.indexOf(charName) != -1) {
                PrevGood = charName;
                continue;
            }

            int MDistance = Integer.MAX_VALUE;
            Character Best = null;

            for (char Good : GoodStr.toCharArray()) {
                int distance = Math.abs(charName - Good);

                if (distance < MDistance) {
                    MDistance = distance;
                    Best = Good;
                } else if (distance == MDistance) {
                    if (Math.abs(PrevGood - Good) < Math.abs(PrevGood - Best)) {
                        Best = Good;
                    }
                }
            }

            Distance += MDistance;
            PrevGood = Best;
        }
        return Distance;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String name = scanner.nextLine();
        String GoodStr = scanner.nextLine();
        int Distance = totalDist(name, GoodStr);
        System.out.println(Distance);
        scanner.close();
    }
}

File: CODEVITA  TCS/Good_String.py
def totalDist(name, GoodStr):
    Distance = 0
    PrevGood = GoodStr[0]

    for char in name:
        if char in GoodStr:
            PrevGood = char
            continue
        
        MDistance = float('inf')
        Best = None
        
        for Good in GoodStr:
            distance = abs(ord(char) - ord(Good))
            
            if distance < MDistance:
                MDistance = distance
                Best = Good
            elif distance == MDistance:
                if abs(ord(PrevGood) - ord(Good)) < abs(ord(PrevGood) - ord(Best)):
                    Best = Good
        
        Distance += MDistance
        PrevGood = Best
    return Distance

name = input()
GoodStr  = input()
# name = "Vyom"
# GoodStr  = "(@HR*i{kcQl" 
Distance = totalDist(name, GoodStr)
print(f"{Distance}")